Classes/New Issue/IssueTemplates.swift

```diff
@@ -0,0 +1,233 @@
+                            client: GithubClient(userSession: details.session),
+                            owner: details.owner,
+                            repo: details.repo,
+                            template: template.template,
+                            signature: template.title == "Bug Report" ? .bugReport : .sentWithGitHawk
+                            ) else {
+                                assertionFailure("Failed to create NewIssueTableViewController")
+                                return
+                        }
+                        viewController.delegate = details.delegate
+                        let navController = UINavigationController(rootViewController: viewController)
+                        navController.modalPresentationStyle = .formSheet
+                        details.viewController.present(navController, animated: trueUnlessReduceMotionEnabled)
+                }))
+        }
+
+        alertView.addAction(
+            UIAlertAction(
+                title: NSLocalizedString("Dismiss", comment: ""),
+                style: .cancel,
+                handler: { _ in
+                    alertView.dismiss(animated: trueUnlessReduceMotionEnabled)
+            })
+        )
+        return alertView
+    }
+
+    static func present(
+        withTemplates sortedTemplates: [IssueTemplate],
+        details: IssueTemplateDetails) {
+        if sortedTemplates.count > 0 {
+            // Templates exists...
+            let alertView = IssueTemplateHelper.showIssueAlert(
+                with: sortedTemplates,
+                details: details
+            )
+            details.viewController.present(alertView, animated: trueUnlessReduceMotionEnabled)
+        } else {
+            // No templates exists, show blank new issue view controller
+            guard let viewController = NewIssueTableViewController.create(
+                client: GithubClient(userSession: details.session),
+                owner: details.owner,
+                repo: details.repo,
+                signature: .sentWithGitHawk
+                ) else {
+                    assertionFailure("Failed to create NewIssueTableViewController")
+                    return
+            }
+            viewController.delegate = details.delegate
+            let navController = UINavigationController(rootViewController: viewController)
+            navController.modalPresentationStyle = .formSheet
+            details.viewController.present(navController, animated: trueUnlessReduceMotionEnabled)
+        }
+    }
+}
+
+extension GithubClient {
+
+    private func fetchTemplateFile(
+        owner: String,
+        repo: String,
+        filename: String,
+        completion: @escaping (Result<String>) -> Void
+        ) {
+
+        self.fetchFile(
+            owner: owner,
+            repo: repo,
+            branch: "master",
+            path: ".github/ISSUE_TEMPLATE/\(filename)") { (result) in
```

Comment:
Have we double checked if GitHub is specific about the caps on this? I know previously when I investigated this I had way too many failures causes by the fact GitHub was case insensitive when checking for this!

(This could be different now though!)

Reply:
Also think @BasThomas picked up on the hardcoded branch here, maybe needs to be the default branch?

Super nit: should we move these constants to somewhere together? (The two path constants)

Reply:
Good point. I remember saying it must be under ISSUE_TEMPLATE but ill recheck.

Reply:
It is case insensitive; we should check that.

Reply:
@BasThomas as in you can have .github/issue_template ? Or the current code is case insensitive?

Reply:
As in you can have `.github/issue_template`

Reply:
@BasThomas @Sherlouk I did a bare bones test adding a file under .github/ISSUE_TEMPLATE and then under .github/issue_template. First one worked, the second one did not. I think it may have to be under ISSUE_TEMPLATE unless someone knows of a counter example

Reply:
So for context my logic was coming from [this official article](https://help.github.com/articles/about-issue-and-pull-request-templates/) but is more about the singular templates as apposed to the new pick a template chooser (as I mentioned in the accompanying issue, this is step 2).

> Issue template filenames are not case sensitive, and can have an extension such as .md or .txt.

If you've tested it we might be safe for this iteration, worst case we raise a bug and fix it at a later date! Ship and iterate!

Reply:
Think this get's a bit confusing because it depends on exactly what the repo is doing but I think as per my comment in the accompanying issue ("step 2") is that single issue templates are not case sensitive and have a lot of different formats but potentially the new multi issue templates require the all caps directory?

https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/

Good article above on the topic showing that the markdown files can be in a bunch of places, and I believe are definitely not case sensitive

Reply:
@Sherlouk Ah I think I understand. The actual filename isn't an issue here because we look at all files in the folder and request those filenames, so they will always match perfectly. The only issue would be the ISSUE_TEMPLATE folder as that's the only hardcoded path. I agree that maybe we ship and fix if there's a bug report. Waiting on Ryan for a few things before I remove the WIP and finish up

Reply:
@Sherlouk Just saw the link, so there's a difference between single issues vs multiple issues it appear (As you mentioned above). I still agree that maybe we go with this and as we learn more about what works etc we fix. If we can't find your templates we still show a blank New Issue screen so it won't have negative effects.